provisioner/ibm-s3fs-provisioner_test.go

```diff
@@ -516,29 +538,77 @@ func Test_Provision_AutoDeleteBucketWithoutAutoCreateBucket(t *testing.T) {
 	}
 }
 
+func Test_Provision_SetDefault_AutoCreateBucket_AutoDeleteBucket_BucketName(t *testing.T) {
+	p := getProvisioner()
+	v := getVolumeOptions()
+
+	_, err := p.Provision(v)
+	assert.NoError(t, err)
+}
+
+func Test_Provision_SetDefault_BucketName(t *testing.T) {
+	p := getProvisioner()
+	v := getVolumeOptions()
+
+	_, err := p.Provision(v)
+	assert.NoError(t, err)
+}
+
+func Test_Provision_SetDefault_AutoCreateBucket_AutoDeleteBucket(t *testing.T) {
+	p := getProvisioner()
+	v := getVolumeOptions()
+	v.PVC.Annotations[annotationBucket] = testBucket
+
+	_, err := p.Provision(v)
+	assert.NoError(t, err)
+}
+
 func Test_Provision_AutoDeleteBucketWithNonEmptyBucket(t *testing.T) {
 	p := getProvisioner()
 	v := getVolumeOptions()
 	v.PVC.Annotations[annotationAutoDeleteBucket] = "true"
 	v.PVC.Annotations[annotationAutoCreateBucket] = "true"
+	v.PVC.Annotations[annotationBucket] = testBucket
 
 	_, err := p.Provision(v)
 	if assert.Error(t, err) {
 		assert.Contains(t, err.Error(), "bucket cannot be set when auto-delete is enabled")
 	}
 }
 
+func Test_Provision_AutoDeleteBucketWithEmptyBucket(t *testing.T) {
+	p := getProvisioner()
+	v := getVolumeOptions()
+	v.PVC.Annotations[annotationAutoDeleteBucket] = "true"
+	v.PVC.Annotations[annotationAutoCreateBucket] = "true"
+
+	_, err := p.Provision(v)
+	assert.NoError(t, err)
+}
+
 func Test_Provision_MissingBucket(t *testing.T) {
 	p := getProvisioner()
 	v := getVolumeOptions()
-	v.PVC.Annotations[annotationBucket] = ""
+	v.PVC.Annotations[annotationAutoDeleteBucket] = "false"
```

Comment:
As per this testcase, no need to set auto-delete-bucket as its already by default set to false.